cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1874,6 +1879,32 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             return withVoid(fut);
         }
 
+        /**
+         * Get the cluster Id of the Kafka cluster
+         * 
+         * @return
+         */
+        Future<ReconciliationState> kafkaGetClusterId() {
+            String clusterId = null;
+            log.debug("{}: Attempt to get clusterId", reconciliation);
+            List<String> podNames = IntStream.range(0, kafkaCluster.getReplicas()).boxed().collect(Collectors.toList()).stream().map(podId -> KafkaCluster.kafkaPodName(this.name, podId)).collect(Collectors.toList());
+            try {
+                String bootstrapHostnames = podNames.stream().map(podName -> KafkaCluster.podDnsName(this.namespace, this.name, podName) + ":" + KafkaCluster.REPLICATION_PORT).collect(Collectors.joining(","));
```

Comment:
You should be able to use the bootstrap service here. That should save you from constructing the name like this.

There is a class `KafkaResources` which has method `bootstrapServiceName` which should be able to give it to you. Or you can add to the `KafkaCluster` class some new method similar to `podDnsName` (since oyu need to use it with the namespace).

I do not think there is any advantage here in using the pod names => it just adds possible complications with DNS changes.

Reply:
ok, I will check that. I did try to use `kafkaBootstrapDnsName` but in my tests it was always empty.